src/transformation/visitors/call.ts

```diff
@@ -290,12 +293,41 @@ export const transformCallExpression: FunctionVisitor<ts.CallExpression> = (node
     return wrapResultInTable ? wrapInTable(callExpression) : callExpression;
 };
 
-function isContextualCallExpression(context: TransformationContext, signature: ts.Signature | undefined): boolean {
-    const declaration = signature?.getDeclaration();
-    if (!declaration) {
-        return !context.options.noImplicitSelf;
+function isContextualCallExpression(
+    context: TransformationContext,
+    calledExpression: ts.Expression,
+    signature?: ts.Signature
+): boolean {
+    const symbol = context.checker.getSymbolAtLocation(calledExpression);
+    let hasNoSelfInFile = false;
+
+    if (symbol?.declarations) {
+        hasNoSelfInFile =
+            symbol.declarations.some(d => getFileAnnotations(d.getSourceFile()).has(AnnotationKind.NoSelfInFile)) ??
+            false;
     }
-    return getDeclarationContextType(context, declaration) !== ContextType.Void;
+
+    if (signature) {
```

Comment:
why did you unwrap the ?. that was there before to this if statement actually? Seems unnecessary